src/models/form.model.ts

```diff
@@ -161,6 +162,38 @@ schema.statics.hasDuplicate = function (
 
 // handle cascading deletion for forms
 addOnBeforeDeleteMany(schema, async (forms) => {
+  for (const form of forms) {
+    if (form.fields && form.fields.length > 0) {
```

Comment:
@ketanreliefapps this is by far too complicated
the ticket is supposed to be simple:
you delete a form, you delete the files, and that's it
no need to check the type of fields etc

Reply:
Done